Tests/CryptoTests/CipherTests.swift

```diff
@@ -42,6 +42,26 @@ class CipherTests: XCTestCase {
         try XCTAssertEqual(AES256.decrypt(ciphertext, key: key).convert(), message)
     }
 
+    func testAES256GCM() throws {
+        let message = "vapor"
+        let key = "passwordpasswordpasswordpassword"
+        let iv = "123456789012"
+        let (ciphertext, tag) = try AES256GCM.encrypt(message, key: key, iv: iv)
+        XCTAssertEqual(ciphertext.hexEncodedString(), "4fa166802c")
+        try XCTAssertEqual(AES256GCM.decrypt(ciphertext, key: key, iv: iv, tag: tag).convert(), message)
+    }
+
+    func testAES256GCMAuthenticationFailure() throws {
+        let message = "vapor"
+        let key = "passwordpasswordpasswordpassword"
+        let iv = "123456789012"
+        let (ciphertext, tag) = try AES256GCM.encrypt(message, key: key, iv: iv)
+        XCTAssertEqual(ciphertext.hexEncodedString(), "4fa166802c")
+
+        let badTag = tag.base64EncodedData()
```

Comment:
Suggestion: use `tag`, but change a single byte. With the current test, decryption might already fail due to an invalid tag length instead of a tag with the correct length, but an invalid value. Might also add a test where `tag` is shortened by a byte (see my comment in the implementation of `setTag`).